### DISCUSSION TITLE
ssh key signatures match exactly, yet auth fails

### DISCUSSION THREAD

**Discussion by danweber** [community/community#36071] [Other Feature Feedback, Questions, & Ideas]
My account's settings page has an ssh key with a signature.

`ssh-add -l` shows the exact same signature:

`256 SHA256:<signature> <account-email> (ED25519)`

`ssh -vv -T git@github.com` shows that exact signature being sent:

```
debug1: Offering public key: <home-dir>/.ssh/id_ed25519 ED25519 SHA256:<signature> agent
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 51
```

What could possibly be wrong?

**Comment by airtower-luna** [comment]
That's not a signature, it's a hash of the public key, no need to censor it. :wink: 

However, for the actual problem: The except of the log you shared doesn't show the authentication failing, only that your SSH client tries to use it. The response from the server is missing, so it's impossible to say what may be going on. Can you share more? Usually one level of debug (`-v`) is enough.

**Reply by danweber** [reply]
It was for an organization-specific key and I should not have been testing git@github.com.

Debugging was set to show debug3. For completeness here is the output with 6 `-v` added. According to some stackoverflow answers, `receive packet: type 51`  means `SSH_MSG_USERAUTH_FAILURE`. Wish ssh actually said it was an auth failure, but that's not anything github can fix.

```
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 51
debug1: Authentications that can continue: publickey
debug2: we did not send a packet, disable method
debug1: No more authentication methods to try.
git@github.com: Permission denied (publickey).

```
